Bucketeer/Sources/Public/BKTConfig.swift

```diff
@@ -11,60 +11,146 @@ public struct BKTConfig {
         }
-        var backgroundPollingInterval = backgroundPollingInterval
+        var backgroundPollingInterval : Int64 = builder.backgroundPollingInterval ?? Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS
         if backgroundPollingInterval < Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS {
-            logger?.warn(message: "backgroundPollingInterval: \(backgroundPollingInterval) is set but must be above \(Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS)")
+            builder.logger?.warn(message: "backgroundPollingInterval: \(backgroundPollingInterval) is set but must be above \(Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS)")
             backgroundPollingInterval = Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS
         }
-        var eventsFlushInterval = eventsFlushInterval
+        var eventsFlushInterval: Int64 = builder.eventsFlushInterval ?? Constant.DEFAULT_FLUSH_INTERVAL_MILLIS
         if eventsFlushInterval < Constant.MINIMUM_FLUSH_INTERVAL_MILLIS {
-            logger?.warn(message: "eventsFlushInterval: \(eventsFlushInterval) is set but must be above \(Constant.MINIMUM_FLUSH_INTERVAL_MILLIS)")
+            builder.logger?.warn(message: "eventsFlushInterval: \(eventsFlushInterval) is set but must be above \(Constant.MINIMUM_FLUSH_INTERVAL_MILLIS)")
             eventsFlushInterval = Constant.DEFAULT_FLUSH_INTERVAL_MILLIS
         }
 
-        self = BKTConfig(
-            apiKey: apiKey,
-            apiEndpoint: apiEndpointURL,
-            featureTag: featureTag,
-            eventsFlushInterval: eventsFlushInterval,
-            eventsMaxQueueSize: eventsMaxQueueSize,
-            pollingInterval: pollingInterval,
-            backgroundPollingInterval: backgroundPollingInterval,
-            sdkVersion: Version.current,
-            appVersion: appVersion,
-            logger: logger
-        )
+        let eventsMaxQueueSize = builder.eventsMaxQueueSize ?? Constant.DEFAULT_MAX_QUEUE_SIZE
+
+        self.init(apiKey: apiKeyForSDK,
+                  apiEndpoint: apiEndpointURL,
+                  featureTag: featureTag,
+                  eventsFlushInterval: eventsFlushInterval,
+                  eventsMaxQueueSize: eventsMaxQueueSize,
+                  pollingInterval: pollingInterval,
+                  backgroundPollingInterval: backgroundPollingInterval,
+                  sdkVersion: builder.sdkVersion,
+                  appVersion: appVersion,
+                  logger: builder.logger)
+    }
+
+    public class Builder {
+        private(set) var apiKey: String?
+        private(set) var apiEndpoint: String?
+        private(set) var featureTag: String?
+        private(set) var eventsFlushInterval: Int64?
+        private(set) var eventsMaxQueueSize: Int?
+        private(set) var pollingInterval: Int64?
+        private(set) var backgroundPollingInterval: Int64?
+        private(set) var sdkVersion: String = Version.current
```

Comment:
`private(set) var sdkVersion: String = Version.current` 
Auto set the `sdkVersion` here